runtime/lua/vim/shared.lua

```diff
@@ -930,7 +972,15 @@ do
   ---               only if the argument is valid. Can optionally return an additional
   ---               informative error message as the second returned value.
   ---             - msg: (optional) error string if validation fails
-  function vim.validate(opt)
+  --- @overload fun(name: string, val: any, expected: string, optional?: boolean)
+  function vim.validate(opt, ...)
+    if select('#', ...) >= 2 then
+      -- Overloaded signature for fast/simple cases
+      local name = opt --[[@as string]]
+      local v, expected, optional = ... ---@type string, string, boolean?
+      return expecttype(name, v, expected, optional)
```

Comment:
is there a reason not to inline `expecttype` ? the extra names feel like needless indirection

Reply:
c9e0debe156fa0551d53c45f66d60d1ca3546400